git-remote-mediawiki

```diff
@@ -504,6 +504,10 @@ sub download_mw_mediafile {
 		# "Wide character in subroutine entry".
 		$response->decode();
 		return $response->content();
+	} elsif (($download_url !~ /^\Q$url\E/) and ($download_url =~ m{\Q$wiki_name\E/})) {
+		# Retry with something based on our original $url
```

Comment:
The commit message is very good, but the comment here would deserve a bit more details IMHO.

Reply:
```suggestion
		# Retry with the path appended onto our original $url
```
or do you prefer something more verbose, like what's in the commit message?

Reply:
(Sorry, I missed the notification for this message :-\)

I'd like to have an explanation on why this is needed (more often than not, a comment that says *what* the patch does is much less useful than one saying *why* it does it). Not necessarily as detailed as the commit message, but at least a hint to the reader.

Reply:
Ah, I see. I suggest this instead:

```suggestion
		# We may have failed because the URL returned from the API
		# is missing something (e.g. "user:password@"). Retry with a
		# corresponding URL constructed from our original $url.
```

Reply:
Perfect, thanks.